src/app/data-model/FabricScopedPreEncodedValue.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "FabricScopedPreEncodedValue.h"
+#include <lib/support/CodeUtils.h>
+
+#include <optional>
+
+namespace chip {
+namespace app {
+namespace DataModel {
+
+FabricScopedPreEncodedValue::FabricScopedPreEncodedValue(const ByteSpan & aData)
+{
+    mReader.Init(aData);
```

Comment:
EncodeForRead will not work if called multiple times, because the internal TLVReader is stateful.

Please keep a copy of the span in the object, rather than just initializing the reader.

Reply:
All operations that involve actually mutating the the reader make a copy of the reader, no?

Reply:
Anyway, resolved.